recipes/qt5/bld.bat

```diff
@@ -1,104 +1,137 @@
 @echo on
-set ENABLEDELAYEDEXPANSION
+setlocal EnableDelayedExpansion
 set QMAKESPEC=win32-msvc%VS_YEAR%
 set SHORT_VERSION=%PKG_VERSION:~0,-2%
 
-:: NOTE: you must have the DirectX SDK installed for ANGLE to compile:
-::    https://www.microsoft.com/en-us/download/details.aspx?id=6812
-::    (572 MB, not availabe through Conda)
-::   You should also set the DXSDK_DIR environment variable once you have the SDK installed.
-
-IF "%DXSDK_DIR%" == "" (
-    echo You do not appear to have the DirectX SDK installed.  Please get it from
-    echo    https://www.microsoft.com/en-us/download/details.aspx?id=6812
-    echo and try this build again.  If you have installed it, and are still seeing
-    echo this message, please open a new console to refresh your environment variables.
-    exit 1
+:: conda-forge/obviousci's obvci_appveyor_python_build_env.cmd used
+:: so we shouldn't include it finally.  I use it to set the correct
+:: Windows SDK, but conda-build should take care of that probably??
+:: It requires TARGET_ARCH to be one of x64 or x86.  On conda-forge
+:: this script is run *before* bld.bat.
+if "%ARCH%" == "64" (
+  set TARGET_ARCH=x64
+) else (
+  set TARGET_ARCH=x86
+)
+call %RECIPE_DIR%\obvci_appveyor_python_build_env.cmd
+
+if "%DXSDK_DIR%" == "" (
+  echo You do not appear to have the DirectX SDK installed.  Please get it from
+  echo    https://www.microsoft.com/en-us/download/details.aspx?id=6812
+  echo and try this build again.  If you have installed it, and are still seeing
+  echo this message, please open a new console to refresh your environment variables.
+  exit /b 1
 )
 
+:: Webengine requires either working OpenGL drivers or
+:: Angle (therefore DirectX >= 11). This works on some
+:: VMs and not others.  Windows 7 VirtualBox instantly
+:: aborts when loading Spyder sSo we've had to disable
+:: it globally.
+:: Using Mesa from MSYS2 is mentioned as a workaround:
+::  http://wiki.qt.io/Cross-compiling-Mesa-for-Windows
+:: `set QT_OPENGL=software` forces Qt5 to use that but
+:: but when I tried it it was too buggy; Spyder crashed a
+:: little bit later, though it worked for Carlos.
+set AVOID_WEBENGINE=yes
+set WEBTHING=qtwebengine
+if %VS_MAJOR% LSS 14 (
+  set WEBTHING=qtwebkit
+)
+if "%AVOID_WEBENGINE%" == "yes" (
+  set WEBTHING=qtwebkit
```

Comment:
I actually built a Qt5 which used both by mistake using a WIP version of this file, so I don't think we want to be mutex here finally, what do you think? It would be nice if we could be dynamic with it in the future, so I'll probably make two vars, `USE_WEBKIT` and `USE_WEBENGINE`.